display/ssd1306.h

```diff
@@ -231,8 +235,16 @@ class SSD1306 : public I2CDevice, Looper, StateMachine, SaberBase, private Audio
 	}
 	frame_count_++;
 	if (looped_frames_ > 1 && millis() - loop_start_ > 5000) {
-	  STDOUT << "END OF LOOP\n";
-	  screen_ = SCREEN_PLI;
+    if (!SaberBase::IsOn()) {
+      STDOUT << "END OF LOOP\n";
+      screen_ = SCREEN_PLI;
+    } else {
+      if (!SaberBase::Lockup()) {
+        ShowFile(&IMG_on);
```

Comment:
I think we need to have some different behavior if "on" is looped or if it's not looped.
If it's not looped, we should just play it once I think.
If it is looped, we should change the 5000 above so that it keeps looping forever.

Right now, I think what happens is that it will play "on" over and over again, regardless of whether it is looped or not. And if it is looped, it will stop the loop after 5 seconds, only to re-start it again.

Reply:
Ok, I've got that all sorted now.

Looped animations for for 'on' will play continuously, single frames will play for 5 seconds.  Same for lockup (as long as the saber is in lockup, that is.)

clash and blast animations will play either for the duration of a single loop OR for 5 seconds in the case of a static image.